src/backend/distributed/transaction/backend_data.c

```diff
@@ -779,6 +781,31 @@ MarkCitusInitiatedCoordinatorBackend(void)
 	SpinLockRelease(&MyBackendData->mutex);
 }
 
+void
+AssignGlobalProcessId(void)
+{
+	if (strncmp(application_name, "citus", 5) != 0)
+	{
+		return;
+	}
+
+	StringInfo applicationName = makeStringInfo();
+	appendStringInfoString(applicationName, application_name);
+
+	char *globalPIDString = &applicationName->data[6];
+	uint64 globalPID = strtoul(globalPIDString, NULL, 10);
+
+	globalPID += 123;
```

Comment:
what's 123?

Reply:
This was just for debugging 😄, removed it